pkg/lib/credentials.js

```diff
@@ -180,78 +180,78 @@ function Keys() {
         return cockpit.script('dir=$(dirname "$1"); test -e "$dir" || mkdir -m 700 "$dir"', [file]);
     }
 
-    function run_keygen(file, new_type, old_pass, new_pass, two_pass) {
+    async function run_keygen(file, new_type, old_pass, new_pass, two_pass) {
         const old_exps = [/.*Enter old passphrase: $/];
         const new_exps = [/.*Enter passphrase.*/, /.*Enter new passphrase.*/, /.*Enter same passphrase again: $/];
         const bad_exps = [/.*failed: passphrase is too short.*/];
 
-        return new Promise((resolve, reject) => {
-            let buffer = "";
-            let sent_new = false;
-            let failure = _("No such file or directory");
+        let buffer = "";
+        let sent_new = false;
+        let failure = _("No such file or directory");
+
+        if (new_pass !== two_pass)
+            return Promise.reject(new Error(_("The passwords do not match.")));
```

Comment:
That looks wrong.  This is an async function, so `return Promise.reject()` here means to return a promise that will *resolve* to a promise which rejects.

The only way you got away with this is because you didn't specify an explicit return type (this is JS, after all...).  I bet if you look at the inferred type it's not going to be what you expect.

I think what you're trying to say is `throw new Error...`.  In async functions, that's a promise rejection.

Reply:
...maybe a testcase to cover this would be nice, too ;)

Reply:
I was nervous about that -- I wasn't sure if an async function *defers* a `throw` before the first call to `await`. But I tested that, and it does work fine. Returning a `Promise.reject()` actually works fine as well, but that's indeed more magical.

I'll create a test case.

Reply:
This isn't actually reachable from the UI -- all calls to this function supply the same variable to `new_pass` and `two_pass`. I'll clean that up instead.

Reply:
Fixed, and removed.